### PR TITLE
implement persistent pool queue service 

### PR DESCRIPTION
**Pull request by mabiede**
- [x] implement persistent pool queue service 
- [x] update email and text messages for the new queue
- [x] switch "MatchesFilter" to new pool queue
- [x] remove old queue

### CODE REVIEW

**Review by timohuber — CHANGES_REQUESTED** [review]

File: pool/app/pool_database/migrations/migration_202407171415.ml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+let drop_queue_jobs_table =
+  Database.Migration.Step.create
+    ~label:"remove sihl queue table"
+    {sql| DROP TABLE IF EXISTS queue_jobs |sql}
+;;
+
+let drop_migration_state_from_sihl_queue =
+  Database.Migration.Step.create
+    ~label:"drop migration state of sihl queue"
+    {sql| DELETE FROM core_migration_state WHERE namespace = 'queue' |sql}
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/pool_database/migrations/migration_202407171415.ml:59]: Yes, the migrations aren't registered anymore, which results in a missing namespace.
As the namespace is unique and we decided to use the timestamps for migrations in future, I'd use it like this.

File: pool/app/pool_database/migrations/migration_202407171415.ml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+let archive_queue_jobs =
+  Database.Migration.Step.create
+    ~label:"archive sihl job queue"
+    {sql|
+      INSERT INTO pool_queue_jobs_history
+        (uuid, name, input, message_template, tries, max_tries, run_at, status,
+        persisted_at, handled_at, last_error, last_error_at, database_label, created_at, updated_at)
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/pool_database/migrations/migration_202407171415.ml:7]: fixed

File: pool/app/email/event.ml [hunk 3]
```diff
@@ -48,18 +49,67 @@ let pp_verification_event formatter (event : verification_event) : unit =
   | EmailVerified m -> pp formatter m
 ;;
 
+type job =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/email/event.ml:52]: this is still open, working on it - depending on the timing I'll create a follow-up issue

[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/email/event.ml]: done

File: pool/app/text_message/event.ml [hunk 4]
```diff
@@ -1,13 +1,43 @@
+open Ppx_yojson_conv_lib.Yojson_conv.Primitives
 open Entity
 
+type job =
```
Comment (timohuber) [inline pool/app/text_message/event.ml:4]: The naming should be updated here as well

File: pool/app/text_message/event.ml [hunk 5]
```diff
@@ -1,13 +1,43 @@
...
+;;
+
 type event =
-  | Sent of job
+  | Sent of (job * Pool_user.CellPhone.t option)
   | BulkSent of job list
   | ReportCreated of delivery_report
 [@@deriving eq, show, variants]
 
+let sent ?new_recipient job = Sent (job, new_recipient)
```
Comment (timohuber) [inline pool/app/text_message/event.ml:22]: `CCPair.make` would do the same, no?

**Review by timohuber — CHANGES_REQUESTED** [review]

File: pool/app/pool_database/root.ml [hunk 6]
```diff
@@ -16,9 +17,12 @@ let steps =
       ; Migration_202312200952.migration ()
       ; Migration_202401081554.migration ()
       ; Migration_202401221730.migration ()
+      ; Migration_202402091434.migration ()
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/pool_database/root.ml:20]: to be able to use the same migration for the root and the tenant databases

**Review by timohuber — APPROVED** [review]

**Pull request merged**